packages/ui-concerto/src/lib/components/concertoForm.js

```diff
@@ -13,213 +13,156 @@
+  const [loading, setLoading] = useState(true);
+  const [modelManager, setModelManager] = useState(null);
+  const { onValueChange } = props;
+
+  const onFieldValueChange = useCallback((e, key) => {
+    const fieldValue = e.type === 'checkbox' ? e.checked : e.value;
+    const valueClone = set({ ...value }, key, fieldValue);
+    setValue(valueClone);
+    onValueChange(valueClone);
+  }, [onValueChange, value]);
+
+  const removeElement = useCallback((e, key, index) => {
+    const array = get(value, key);
     array.splice(index, 1);
-    this.props.onValueChange(this.state.value);
-  }
+    onValueChange(value);
+  }, [onValueChange, value]);
 
-  addElement(e, key, value) {
-    const array = get(this.state.value, key) || [];
+  const addElement = useCallback((e, key, elementValue) => {
+    const array = get(value, key) || [];
     const valueClone = set(
-      { ...this.state.value },
+      { ...value },
       [...key, array.length],
-      value
+      elementValue
     );
-    this.setState({ value: valueClone });
-    this.props.onValueChange(valueClone);
-  }
-
-  isInstanceOf(model, type) {
-    return this.generator.isInstanceOf(model, type);
-  }
+    setValue(valueClone);
+    onValueChange(valueClone);
+  }, [onValueChange, value]);
+
+  // Default values which can be overridden by parent components
+  const options = React.useMemo(() => ({
+    includeOptionalFields: true,
+    includeSampleData: 'sample',
+    disabled: props.readOnly,
+    visitor: new ReactFormVisitor(),
+    onFieldValueChange: (e, key) => {
+      onFieldValueChange(e, key);
+    },
+    addElement: (e, key, field) => {
+      addElement(e, key, field);
+    },
+    removeElement: (e, key, index) => {
+      removeElement(e, key, index);
+    },
+    ...props.options,
+  }), [addElement, onFieldValueChange, removeElement, props.options, props.readOnly]);
+
+  const generator = React.useMemo(() => {
+    if (modelManager) {
+      return new FormGenerator(modelManager, options);
+    }
+    return null;
+  }, [modelManager, options]);
+
+  useEffect(() => {
+    setLoading(true);
+    const modelManager = new ModelManager();
+    // TODO Refactor this to an option to make this independent of Cicero
+    modelManager.addModelFile(
+      `namespace org.accordproject.base
+    abstract asset Asset {  }
+    abstract participant Participant {  }
+    abstract transaction Transaction identified by transactionId {
+      o String transactionId
+    }
+    abstract event Event identified by eventId {
+      o String eventId
+    }`,
+      'org.accordproject.base.cto',
+      false,
+      true
+    );
+    props.models.forEach((model, idx) => {
+      const decodedModel = decodeHTMLEntities(model);
+      modelManager.addModelFile(decodedModel, `model-${idx}`, true);
+    });
+    modelManager.updateExternalModels().then(() => {
```

Comment:
I think this we used to have an option to disable the download of external models?